Riot/Modules/Room/RoomViewController.m

```diff
@@ -7276,6 +7501,54 @@ - (void)documentPickerPresenter:(MXKDocumentPickerPresenter *)presenter didPickD
     }
 }
 
+- (void)sendImage:(NSData *)imageData mimeType:(NSString *)mimeType {
+    // Re-invite the left member before sending the message in case of a discussion (direct chat)
+    MXWeakify(self);
+    [self createOrRestoreDiscussionIfNeeded:^(BOOL success) {
+        MXStrongifyAndReturnIfNil(self);
+        if (success)
+        {
+            // Let the datasource send it and manage the local echo
+            [self.roomDataSource sendImage:imageData mimeType:mimeType success:nil failure:^(NSError *error) {
+                // Nothing to do. The image is marked as unsent in the room history by the datasource
+                MXLogDebug(@"[MXKRoomViewController] sendImage failed.");
+            }];
+        }
+    }];
+}
+
+- (void)sendVideo:(NSURL * _Nonnull)url {
+    // Re-invite the left member before sending the message in case of a discussion (direct chat)
+    MXWeakify(self);
+    [self createOrRestoreDiscussionIfNeeded:^(BOOL success) {
+        MXStrongifyAndReturnIfNil(self);
+        if (success)
+        {
+            // Let the datasource send it and manage the local echo
+            [(RoomDataSource*)self.roomDataSource sendVideo:url success:nil failure:^(NSError *error) {
```

Comment:
It this cast really needed ? There is no cast for image and file (only for video)

Reply:
After another review, I fount it is really needed.